### DISCUSSION TITLE
Issue with size of the action menu or other menus

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#163] [Q&A] [answered]
There are some situations, can't exactly tell when, where the menu size is too small in portait mode. When you then turn the phone to landscape mode, the menu is full size again, also if you turn back to portrait mode. 
Have no clue what could be the reason, screenshot here

<img src="https://github.com/devemux86/cruiser/assets/121371629/a45893c1-9876-45d7-aaf3-192bf802dda5" width="40%"/> <img src="https://github.com/devemux86/cruiser/assets/121371629/5f7beac4-f889-48f1-a375-a31ade6a68a4" width="40%"/>

**Comment by devemux86** [comment]
@DolomitesTom thanks for the report.

This can probably happen when open the menus in a device orientation (portrait / landscape),
change the device orientation and open the menus again.

Although they are created again in the new device orientation,
Android seems to keep something from the previous orientation.

I will check it!

**Reply by MotoUKRider** [reply]
I often see this if I get "creative" with my rotating during my (now frequent!) attempts to break Cruiser.

Although not obvious, the compacted menu does indeed "scroll" :)

**Comment by devemux86** [comment]
> (now frequent!) attempts to break Cruiser

:+1: 
I welcome such efforts!

Please try to break Cruiser, use it in unusual ways and push it to its limits!
This is a good way to discover any issues and make Cruiser even better. :slightly_smiling_face:

**Comment by devemux86** [comment]
It will work properly in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
It should work properly with [Cruiser 3.0.21](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by DolomitesTom** [comment]
![Screenshot_20230607-060054_Cruiser](https://github.com/devemux86/cruiser/assets/121371629/4008989d-6e62-4b9a-a2d8-70c4de31e4b0)


Have this issue still or again in the context menu when long pressing the route.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the report!

- Can you share the app settings `.pref` file?
- Can you also mention the steps this happens?

e.g. Do you open the dialog after app start in landscape, rotate the phone in portrait and open the dialog again?
This should be fixed in latest Cruiser.

**Reply by DolomitesTom** [reply]
pref file sent by email

I open in portrait mode then this happens when just pressing the menu
![grafik](https://github.com/devemux86/cruiser/assets/121371629/bb073d53-d4a5-47c9-aef8-99e5f8b3f8ac)

when turning the phone into landscape and then back to portait again, then the menu is enlarged again.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the .pref file.

It is strange, loading your .pref file, the menu appears properly like in the image.

You have changed the font + display size in application **advanced** settings.
Can you revert these 2 settings to System or 100% and try again?

> I open in portrait mode then this happens when just pressing the menu

Do you mean after the app starts for the 1st time?
What happens if you remove the app from memory with Android recents button?

<img src="https://github.com/devemux86/cruiser/assets/3484020/ac6e5f0a-67fa-47ca-94d7-9942a3913364" width="50%"/>

**Reply by DolomitesTom** [reply]
after closing, swiping the ram and restarting Cruiser, the menu is normal, also with font + display size in application advanced settings, but this is too small for me. Maybe the problem is when the app stays a while in memory.

**Reply by devemux86** [reply]
> Maybe the problem is when the app stays a while in memory.

Probably, but I need the detailed steps that this happens, so I can reproduce and improve it.
You first report [above](https://github.com/devemux86/cruiser/discussions/163#discussion-5190683) could be reproduced, so I could find the fix.

By the way, next Cruiser will also offer a separate size setting for the screen buttons.
After that, are the font + display size in application **advanced** settings needed?
Aren't they becoming redundant? As they can usually break the app UI.

**Reply by DolomitesTom** [reply]
don't know the detailed steps, will try to recall when it happens again, at the moment it is working. 

I will try it with the next version  and with  font + display default settings, but please don't delete it before. If it is working, it could be deleted with another version.

**Reply by devemux86** [reply]
> I will try it with the next version and with font + display default settings, but please don't delete it before. If it is working, it could be deleted with another version.

It can remain, if we are comfortable without it, we can examine the possibility. :slightly_smiling_face: 

That's why I recommend everyone to try only the font + display settings in "Appearance" screen.
(without changing the advanced sizes)

@DolomitesTom have you changed the font + display size in **Android display settings**?

**Reply by MotoUKRider** [reply]
> That's why I recommend everyone to try only the font + display settings in "Appearance" screen. (without changing the advanced sizes)

These settings have no effect that I can see (Samsung S20).  "Advanced" works fine, but creates issues with menu sizing in landscape (as per other thread), so while the next update will have the on-map (round) button sizes as a new setting, what *should* the Appearance Display+Font sliders alter?

**Reply by devemux86** [reply]
@MotoUKRider please see my answer here: https://github.com/devemux86/cruiser/discussions/129#discussioncomment-6110833

**Reply by DolomitesTom** [reply]
> have you changed the font + display size in Android display settings?

now yes to 130%. But this doesn't change the font size of the menu texts

**Reply by devemux86** [reply]
> But this doesn't change the font size of the menu texts

The menu buttons?
These have large icons, we should rely on the icons. :slightly_smiling_face: 

Or you mean the text menus, like when long press on the map?
